src/ext/java/org/opentripplanner/ext/flex/trip/UnscheduledTrip.java

```diff
@@ -68,16 +71,30 @@ public static UnscheduledTripBuilder of(FeedScopedId id) {
     return new UnscheduledTripBuilder(id);
   }
 
+  /**
+   * Tests if the stop times constitute an {@link UnscheduledTrip}.
+   * <p>
+   * Returns true for the following cases:
+   *  - A single fixed scheduled stop followed by a flexible one
+   *  - One or more stop times with a flexible time window but no fixed stop in between them
+   */
   public static boolean isUnscheduledTrip(List<StopTime> stopTimes) {
     Predicate<StopTime> hasFlexWindow = st ->
       st.getFlexWindowStart() != MISSING_VALUE || st.getFlexWindowEnd() != MISSING_VALUE;
     Predicate<StopTime> notContinuousStop = stopTime ->
       stopTime.getFlexContinuousDropOff() == NONE && stopTime.getFlexContinuousPickup() == NONE;
-    return (
-      N_STOPS.contains(stopTimes.size()) &&
-      stopTimes.stream().anyMatch(hasFlexWindow) &&
-      stopTimes.stream().allMatch(notContinuousStop)
-    );
+    boolean noContinuousStops = stopTimes.stream().allMatch(notContinuousStop);
+    if (stopTimes.isEmpty()) {
+      return false;
+    } else if (N_STOPS.contains(stopTimes.size())) {
+      return (
+        N_STOPS.contains(stopTimes.size()) &&
+        stopTimes.stream().anyMatch(hasFlexWindow) &&
+        noContinuousStops
+      );
+    } else {
+      return stopTimes.stream().allMatch(hasFlexWindow) && noContinuousStops;
+    }
```

Comment:
Something like this is a bit more readable:

```
    if (stopTimes.isEmpty()) {
      return false;
    }
    if(stopTimes.stream().anyMatch(continuousStop)) {
      return false;
    }
    if (N_STOPS.contains(stopTimes.size())) {
      return stopTimes.stream().anyMatch(hasFlexWindow);
    }
    return stopTimes.stream().allMatch(hasFlexWindow);
```

Reply:
Good point. Changed in bb38a6c3d62ebcf7b64db26a93dd9f48967a3404